### PR TITLE
Add tests for umitools extract and dedup

### LINKED ISSUE #1425: [BUG] umitools dedup has no tests
Follow up to #1422

The module was created before we inforced any testing. We should probably add linting to tools to make sure that there's at least test files there, since there was no failure @grst just noticed that pytest workflow didn't kick off.

### PR DESCRIPTION
**Pull request by SPPearce**
## PR checklist

Closes #1425, #1254 and #108.
Umitools had no tests, so I've added some. I've also made the dedup command deterministic, by fixing the random seed option. That isn't currently tested, as it only occurs when the bam file has actual duplicated reads.

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [X] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/umitools/dedup/main.nf [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+include { BWA_MEM   } from '../../../../modules/bwa/mem/main.nf'
+include { SAMTOOLS_INDEX   } from '../../../../modules/samtools/index/main.nf'
+include { UMITOOLS_DEDUP } from '../../../../modules/umitools/dedup/main.nf'
+
+//
+// Test with single-end data
+//
+workflow test_umitools_dedup_single_end {
+    input = [ [ id:'test', single_end:true ], // meta map
+              [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) ]
```
Comment (Emiller88) [inline tests/modules/umitools/dedup/main.nf:16]: Could we just use the test bams and bais for this? Or are they already deduped (I don't know exactly what the dedup is doing)? If so, could we add a comment about why the extra steps are included?

Maybe we could keep these to just dedup. Maybe we can keep the e2e tests(which are a good example of how to use the modules), but the minimal tests remove the dependencies on BWA and samtools changing under the hood. I can write up some of those if I'm not being clear.

**Review by Emiller88 — CHANGES_REQUESTED** [review]

File: tests/modules/umitools/dedup/main.nf (hunk 1 above)
Comment (SPPearce) [reply tests/modules/umitools/dedup/main.nf:16]: But I do like showing how to actually use the module...

Comment (Emiller88) [reply tests/modules/umitools/dedup/main.nf:16]: > The test bams don't have unique molecular identifiers (UMIs) present.


That's what I was guessing!

> But I do like showing how to actually use the module...


 ¿por que no los dos?
 
 Let's just have both tests. Just add the `--ignore-umi` flag

File: tests/config/pytest_modules.yml [hunk 2]
```diff
@@ -1757,6 +1757,14 @@ ultra/pipeline:
   - modules/ultra/pipeline/**
   - tests/modules/ultra/pipeline/**
 
+umitools/dedup:
+  - modules/umitools/dedup/**
```
Comment (Emiller88) [inline tests/config/pytest_modules.yml:1761]:
```suggestion
  - modules/umitools/dedup/**
  - modules/umitools/extract/**
```

To catch regressions.

File: tests/modules/umitools/dedup/main.nf [hunk 3]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { UMITOOLS_EXTRACT } from '../../../../modules/umitools/extract/main.nf'
+include { BWA_INDEX } from '../../../../modules/bwa/index/main.nf'
+include { BWA_MEM   } from '../../../../modules/bwa/mem/main.nf'
+include { SAMTOOLS_INDEX   } from '../../../../modules/samtools/index/main.nf'
+include { UMITOOLS_DEDUP } from '../../../../modules/umitools/dedup/main.nf'
```
Comment (Emiller88) [inline tests/modules/umitools/dedup/main.nf:9]:
```suggestion
include { 
    UMITOOLS_DEDUP 
    UMITOOLS_DEDUP as UMITOOLS_DEDUP_NOUMI 
} from '../../../../modules/umitools/dedup/main.nf'
```

**Review by Emiller88 — CHANGES_REQUESTED** [review]
Just a few things.

**Review by Emiller88 — APPROVED** [review]
LGTM!

**Pull request merged**